### PR TITLE
Diagonal background animation

### PR DESCRIPTION
**Pull request by tushar2407**
Issue:  [BUTTON IDEA]: Diagonally appearing background on hover #482

### CODE REVIEW

**Comment by tushar2407** [comment]
Hi @shahednasser,
I have implemented my idea. Hope you like it. 
I am having trouble changing the colour of the yellow button. If you could please help me with the same.

**Comment by HADES-01** [comment]
@tushar2407 Great Work. Please change class name to `diag-btn` or something smaller. And about the yellow button issue, There's this piece of code in src/components/_basic.less:
```CSS
.yellow-btn {
  .colorRule(@yellow);

  &:not(.fill-color-btn):not(.ripple-btn) {
    color: @darkText !important;
  }
}
```
just change this ` &:not(.fill-color-btn):not(.ripple-btn)` to ` &:not(.fill-color-btn):not(.ripple-btn):not(.your-class-name)`

File: assets/js/buttons-examples.js [hunk 1]
```diff
@@ -154,6 +154,11 @@ $(document).ready(function () {
           label: "Transparent On Hover",
           classes: "transparent-btn",
         },
+        {
```
Comment (shahednasser) [inline assets/js/buttons-examples.js:157]: Please place the button in the correct alphabetical order

**Review by shahednasser — CHANGES_REQUESTED** [review]
In addition to what @HADES-01 mentioned please fix the white button as the text is turning white on hover (should be black)

File: assets/js/buttons-examples.js (hunk 1 above)
Comment (tushar2407) [reply assets/js/buttons-examples.js]: Sure done.
Thanks

**Comment by tushar2407** [comment]
> @tushar2407 Great Work. Just change the class name to `diag-btn` or something smaller.
> 
> And about the yellow button issue, There's this piece of code in src/components/_basic.less:
> 
> ```css
> .yellow-btn {
>   .colorRule(@yellow);
> 
>   &:not(.fill-color-btn):not(.ripple-btn) {
>     color: @darkText !important;
>   }
> }
> ```
> 
> just change this ` &:not(.fill-color-btn):not(.ripple-btn)` to ` &:not(.fill-color-btn):not(.ripple-btn):not(.your-class-name)`

Thanks @HADES-01. 
Changes are done.

File: src/components/animated/_diagonalBackground.less [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  }
+  &.white-btn {
+    background-image: linear-gradient(
+      120deg,
+      transparent 50%,
+      @white 50%
+    ) !important;
+    color: #313131 !important;
+  }
+  &:hover:not(.white-btn) {
```
Comment (shahednasser) [inline src/components/animated/_diagonalBackground.less:82]: can you add to this `:not(.yellow-btn)`? The text for the yellow button on hover changes to white but it should change to black.

**Review by shahednasser — CHANGES_REQUESTED** [review]

File: src/components/animated/_diagonalBackground.less (hunk 2 above)
Comment (tushar2407) [reply src/components/animated/_diagonalBackground.less]: @shahednasser Done.

**Comment by shahednasser** [comment]
Great work thank you!

@all-contributors please add @tushar2407 for code

**Pull request merged**